controllers/userStatus.js

```diff
@@ -84,28 +92,36 @@ const getAllUserStatus = async (req, res) => {
  */
 const updateUserStatus = async (req, res) => {
   try {
-    const { userId } = req.params;
-    const dataToUpdate = req.body;
-    const updateStatus = await userStatusModel.updateUserStatus(userId, dataToUpdate);
-    const { userStatusExists, userStatusUpdated, id, data } = updateStatus;
-    let responseObject = { id, userId };
-    let statusCode;
-    if (data) {
-      responseObject = { ...responseObject, ...data };
-    }
-    if (userStatusExists) {
-      responseObject.message = "User Status updated successfully.";
-      statusCode = 200;
+    let userId;
+    if (req.route.path === "/self") {
+      userId = req.userData.id;
     } else {
-      if (userStatusUpdated) {
-        responseObject.message = "User Status created successfully.";
-        statusCode = 201;
+      userId = req.params.userId;
+    }
+    if (userId) {
+      const dataToUpdate = req.body;
+      const updateStatus = await userStatusModel.updateUserStatus(userId, dataToUpdate);
+      const { userStatusExists, userStatusUpdated, id, data } = updateStatus;
+      let responseObject = { id, userId };
+      let statusCode;
+      if (data) {
+        responseObject = { ...responseObject, ...data };
+      }
+      if (userStatusExists) {
+        responseObject.message = "User Status updated successfully.";
+        statusCode = 200;
       } else {
-        responseObject.message = "User Status couldn't be created due to incomplete request body.";
-        statusCode = 400;
+        if (userStatusUpdated) {
+          responseObject.message = "User Status created successfully.";
+          statusCode = 201;
+        } else {
+          responseObject.message = "User Status couldn't be created due to incomplete request body.";
+          statusCode = 400;
+        }
```

Comment:
Same `responseObject` schema consistency.

Reply:
I agree with your point but since I am not adding `data` as a direct key here instead I am destructuring the `data` object so that all the keys of `data` become the primary keys of `responseObject` as well. So if the `data` object exist then only those keys will be added if not then we are adding a relevant message `User Status couldn't be found.`